### DISCUSSION TITLE
When / Why are some eloquent models cast to array and some arent? What to do about it?

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#4023] [1. Help]
issue is>
sometimes an eloquent model in a livewire component gets cast to array. Not always though.

1. What is the condition that that happens
2. What can I do about it to avoid it?

i read as well here:
https://github.com/livewire/livewire/issues/27#issuecomment-521040661

**Comment by joshhanley** [comment]
@torian257x hmm that depends on how the eloquent models are loaded, and if raw db calls are made, or `toBase` used on Eloquent Collections.

Do you have an example of a component code you can share that has a model that is being rehydrated as an array? If you do, I can have a look to see what might be causing it.

Hope this helps!

**Reply by torian257x** [reply]
we use a
`->orderByRaw` is that what can cause it?

**Reply by joshhanley** [reply]
@torian257x hmm in theory not, as the returned object should still be an eloquent collection. It's when a query returns a normal collection with models in it, that they will be objects on the first request (as they just came from the database), but arrays on the next request.

Livewire is only smart if it's an eloquent collection, as they will generally always contains eloquent models.
We can't add that to standard collections, as we don't necessarily know what will be in the collection.

Hope this helps!

**Reply by torian257x** [reply]
sooo we have `\App\Models\Orders`

and we do

`Orders::get()->groupBy('status');` where groupBy is NOT a query builder method, but a method of collection, returning a base Collection.

Any recommendation on what to do? I would be very thankful if this would be documented somewhere as well.

**Reply by joshhanley** [reply]
@torian257x ah yep, that would do it. 

In terms of what to do about it, I would probably assign the non-grouped collection to a property (that way it's still an eloquent collection) and then setup a computed property for the grouped collection, like

```php
public $orders;

public function mount() {
    $this->orders = Orders::get();
 }
 
 public function getGroupedOrdersProperty() {
     return $this->orders->groupBy('status');
 }
 ```
 
 Then in your view just use `$this->groupedOrders` to access the grouped collection.
 
 Yeah agreed, docs PR's are always welcome https://github.com/livewire/docs
 
 Hope this helps!
